substrate/primitives/io/src/lib.rs

```diff
@@ -284,10 +302,10 @@ pub trait Storage {
 	#[version(3, register_only)]
 	fn clear_prefix(
 		&mut self,
-		maybe_prefix: &[u8],
-		maybe_limit: Option<u32>,
-		maybe_cursor: Option<Vec<u8>>, //< TODO Make work or just Option<Vec<u8>>?
-	) -> MultiRemovalResults {
+		maybe_prefix: PassFatPointerAndRead<&[u8]>,
+		maybe_limit: PassByCodec<Option<u32>>,
+		maybe_cursor: PassByCodec<Option<Vec<u8>>>, //< TODO Make work or just Option<Vec<u8>>?
```

Comment:
I know this TODO comment has been here before, but is it clear what it means and should it stay?

Reply:
Hmm.... no, I have no idea what it means. :P

This host function is `register_only` which means that it's not used. I tried to search around, but I couldn't find any docs as to whether there are any *concrete* plans to actually use it. It's [not in the spec](https://spec.polkadot.network/chap-host-api#id-ext_storage_clear_prefix) and there's no open RFC for it.

It was originally added in [this PR](https://github.com/paritytech/substrate/pull/11490), and there's this [TODO here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/support/src/storage/unhashed.rs#L145) to eventually enable it.

So either someone actually pushes it forward, or we delete it.

cc @ggwpez @KiChjang Do you guys perhaps know if there are any plans to do anything (= write an RFC) with this host function?